File: ios/FamilyApp/Features/Home/HomeGreetingUtils.swift
// Time-of-day greeting.
import Foundation

private let lastMorningHour = 11
private let lastAfternoonHour = 17

/// "Good morning" (0–11), "Good afternoon" (12–17), or "Good evening" (18–23).
func timeBasedGreeting(
    hour: Int = Calendar.current.component(.hour, from: Date()),
    locale: Locale = Locale(identifier: "en_US_POSIX")
) -> String {
    switch hour {
    case 0...lastMorningHour: L("Good morning", locale: locale)
    case (lastMorningHour + 1)...lastAfternoonHour: L("Good afternoon", locale: locale)
    default: L("Good evening", locale: locale)
    }
}

File: android/app/src/main/java/com/example/mainactivity/ui/home/HomeGreetingUtils.kt
package com.example.mainactivity.ui.home

import java.util.Calendar

private const val LAST_MORNING_HOUR = 11
private const val LAST_AFTERNOON_HOUR = 17

/** Returns "Good morning", "Good afternoon", or "Good evening" based on the hour. */
internal fun timeBasedGreeting(): String =
    when (Calendar.getInstance().get(Calendar.HOUR_OF_DAY)) {
        in 0..LAST_MORNING_HOUR -> "Good morning"
        in (LAST_MORNING_HOUR + 1)..LAST_AFTERNOON_HOUR -> "Good afternoon"
        else -> "Good evening"
    }
